index.d.ts

```diff
@@ -1,12 +1,65 @@
-import { Parser } from 'acorn' 
+import * as acorn from 'acorn';
 
-declare const jsx: (options?: jsx.Options) => (BaseParser: typeof Parser) => typeof Parser;
+declare const jsx: {
+  tokTypes: typeof acorn.tokTypes;
+  (options?: jsx.Options): (BaseParser: typeof acorn.Parser) => jsx.AcornJsxParser
+}
 
 declare namespace jsx {
-  interface Options {
+
+  type tokTypes = typeof acorn.tokTypes;
+
+  export interface TokTypes extends tokTypes {
+    jsxName: acorn.TokenType,
+    jsxText: acorn.TokenType,
+    jsxTagEnd: acorn.TokenType,
+    jsxTagStart: acorn.TokenType
+  }
```

Comment:
I suggest to do the following:

```suggestion
  export interface TokTypes extends typeof acorn.tokTypes {
```

There’s no need to export the `tokTypes` type from acorn. Also it’s generally preferred to use the CamelCase for type names, which isn’t used for `tokTypes`. By not exporting them, this is avoided altogether.

Reply:
The reason I added a type alias was because I get an "expression expected" error. See https://github.com/Microsoft/TypeScript/issues/14757 on why TypeScript isn't supporting this.

Should I move the type alias out of the namespace so it is not exported then?

And if so, would you prefer for CamelCase that I change it to `AcornTokTypes`?

Reply:
Also, and I guess this might have been inadvertent in making the commit suggestion, we don't want to delete the properties in the interface.

Reply:
> The reason I added a type alias was because I get an "expression expected" error if directly adding it as in your suggestion. See [microsoft/TypeScript#14757](https://github.com/microsoft/TypeScript/issues/14757) on why TypeScript isn't supporting this.
> 
> Should I move the type alias out of the namespace so it is not exported then?
> 
> And if so, would you prefer for CamelCase that I change it to `AcornTokTypes`?

I've gone ahead and added a commit to do this. Let me know if this is ok.